[[ Toolchain ]] Allow foreign handlers to be marked as safe

runrevmark opened the pull request:
This patch allows a foreign handler definition to be prefixed by '__safe'
indicating that it is safe to call from safe context.

The motivation for this is to ease binding to C/C++ functions which are
implemented with the LCB C calling convention.

peter-b commented:
What do you think about having an "experimental features" section in the language spec that describes this syntax?  It would be a great shame for this to be completely undocumented anywhere.

peter-b commented:
Could you please add some tests for this syntax?

runrevmark commented:
@peter-b: Yes - I've got some on the __bridged (closed) PR - I'll port across. What do you suggest the structure of an 'experimental features' section should be?

peter-b commented:
Something like:

``` markdown
## Experimental features

**Warning**: This section describes current language features and syntax
that are considered experimental and unstable.  They are likely to change
or go away without warning.
```

runrevmark committed:
lcb: Add safe foreign handler support

This patch allows a foreign handler definition to be prefixed by '__safe'
indicating that it is safe to call from safe context.

The motivation for this is to ease binding to C/C++ functions which are
written to be easily callable from LCB.
